File: app/src/main/java/edu/hm/cs/krivoj/mensapp/menu/MenuActivity.java
package edu.hm.cs.krivoj.mensapp.menu;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.util.Log;

import java.time.LocalDate;

import edu.hm.cs.krivoj.mensapp.R;
import edu.hm.cs.krivoj.mensapp.api.menu.Menu;
import edu.hm.cs.krivoj.mensapp.api.price.Table;

public class MenuActivity extends AppCompatActivity {

    public static final String EXTRA_MENU = "edu.hm.cs.krivoj.mensapp.menu.MenuActivity.menu";
    public static final String EXTRA_PRICETABLE =
            "edu.hm.cs.krivoj.mensapp.menu.MenuActivity.pricetable";
    private static final String LOG_TAG = "MenuActivity";

    private Menu menu;

    private LinearLayoutManager layoutManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);

        menu = (Menu) getIntent().getExtras().getSerializable(EXTRA_MENU);
        Table table = (Table) getIntent().getExtras().getSerializable(EXTRA_PRICETABLE);

        Toolbar toolbar = findViewById(R.id.toolbar);
        toolbar.setTitle(menu.getCanteen().getName());
        setSupportActionBar(toolbar);

        RecyclerView recyclerView = findViewById(R.id.recyclerViewDailyplans);
        layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(new DailyPlanAdapter(menu, table));
        scrollToToday();

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(v -> scrollToToday());
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    private void scrollToToday() {
        Log.d(LOG_TAG, "Scrolling to today");
        LocalDate today = LocalDate.now();
        layoutManager.scrollToPositionWithOffset(menu.indexOf(today), 16);
    }
}

File: app/src/main/java/edu/hm/cs/krivoj/mensapp/menu/MenuActivity.kt
package edu.hm.cs.krivoj.mensapp.menu

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.util.Log
import edu.hm.cs.krivoj.mensapp.R
import edu.hm.cs.krivoj.mensapp.api.Canteen
import edu.hm.cs.krivoj.mensapp.api.menu.Menu
import edu.hm.cs.krivoj.mensapp.api.price.Table
import kotlinx.android.synthetic.main.activity_menu.*
import java.time.LocalDate

class MenuActivity : AppCompatActivity() {

    private var menu: Menu? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)

        menu = intent.extras!!.getSerializable(EXTRA_MENU) as Menu
        val table = intent.extras!!.getSerializable(EXTRA_PRICETABLE) as Table
        val canteen = intent.extras!!.getSerializable(EXTRA_CANTEEN) as Canteen

        toolbar.title = canteen.name
        setSupportActionBar(toolbar)

        recyclerViewDailyplans.layoutManager = LinearLayoutManager(this)
        recyclerViewDailyplans.adapter = DailyPlanAdapter(menu, table)
        scrollToToday()

        fab.setOnClickListener { scrollToToday() }
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    private fun scrollToToday() {
        Log.d(LOG_TAG, "Scrolling to today")
        val today = LocalDate.now()
        val layoutManager = recyclerViewDailyplans.layoutManager as LinearLayoutManager
        layoutManager.scrollToPositionWithOffset(menu!!.indexOf(today), 16)
    }

    companion object {
        const val EXTRA_MENU = "edu.hm.cs.krivoj.mensapp.menu.MenuActivity.menu"
        const val EXTRA_CANTEEN = "edu.hm.cs.krivoj.mensapp.menu.MenuActivity.canteen"
        const val EXTRA_PRICETABLE = "edu.hm.cs.krivoj.mensapp.menu.MenuActivity.pricetable"
        private const val LOG_TAG = "MenuActivity"
    }
}
